### PR TITLE
Add vcpkg manifest installation instructions

### PR DESCRIPTION
**Pull request by FrankXie05**
`Snitch` is available as a port in vcpkg, a C++ library manager that simplifies installation for `Snitch` and other project dependencies. 

This PR provides a set of example project files to install and use `snitch` via vcpkg's mainfest mode.

We also test whether our library ports build in various configurations (dynamic, static) on various platforms (OSX, Linux, Windows: x86, x64, UWP, ARM) to keep a wide coverage for users.

I'm a maintainer for vcpkg, and here is what the port script looks like. We try to keep the library maintained as close as possible to the original library. 😊

Related information: [manifest-mode](https://learn.microsoft.com/en-us/vcpkg/concepts/manifest-mode)

### CODE REVIEW

File: vcpkg-example/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Snitch example
+ 
+Snitch example that uses [vcpkg][vcpkg].
+ 
+## Set up vcpkg
+ 
+Clone the vcpkg and run the bootstrap script.
```
Comment (cschreib) [inline vcpkg-example/README.md:7]:
```suggestion
Clone vcpkg and run the bootstrap script.
```

File: vcpkg-example/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Snitch example
```
Comment (cschreib) [inline vcpkg-example/README.md:1]:
```suggestion
# Example of using *snitch* with vcpkg
```

File: vcpkg-example/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ 
+Snitch example that uses [vcpkg][vcpkg].
```
Comment (cschreib) [inline vcpkg-example/README.md:3]:
```suggestion
This is an example of building a project with *snitch* as a dependency, using [vcpkg][vcpkg].
```

File: vcpkg-example/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+```bash
+git clone https://github.com/microsoft/vcpkg
+cd vcpkg && ./bootstrap-vcpkg.bat
+```
+ 
+## Build and run the project
+
+### CMake
+
+We provide a project folder called `vcpkg-example` to use `snitch`. You can build this example project according to the following code.
```
Comment (cschreib) [inline vcpkg-example/README.md:18]:
```suggestion
The folder containing this README includes CMake files for an example empty project using  *snitch*. You can build this example project with the following code.
```

File: vcpkg-example/main.cpp [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿#include <iostream>
+#include <optional>
```
Comment (cschreib) [inline vcpkg-example/main.cpp:2]: These should not be needed and can be removed.

**Review by cschreib — CHANGES_REQUESTED** [review]
Thanks for contributing this! 

I have left a few minor comments on grammar/presentation/style to match the rest of the documentation. Speaking of which, could you also move this folder into the `doc` folder please, so that all documentation is in the same place?

Lastly, it would probably be great to add a link to this README from the root README, so people can find it. Perhaps another section after "Example build configuration with meson" called "Example build configuration with vcpkg", which just contains a short sentence with a link to your new README?

**Comment by FrankXie05** [comment]
> Thanks for contributing this!
> 
> I have left a few minor comments on grammar/presentation/style to match the rest of the documentation. Speaking of which, could you also move this folder into the `doc` folder please, so that all documentation is in the same place?
> 
> Lastly, it would probably be great to add a link to this README from the root README, so people can find it. Perhaps another section after "Example build configuration with meson" called "Example build configuration with vcpkg", which just contains a short sentence with a link to your new README?

Thanks for the reply, I will finish it right away. ：:)

File: vcpkg-example/main.cpp (hunk 5 above)
Comment (FrankXie05) [reply vcpkg-example/main.cpp:2]: Do you mean we don't need `main.cpp` or just part of the code? Or just a simple `main.cpp`?

Comment (cschreib) [reply vcpkg-example/main.cpp]: Just the two lines I highlighted ;)

**Review by cschreib — APPROVED** [review]

**Comment by cschreib** [comment]
Thank you!

**Pull request merged**